### DISCUSSION TITLE
GetRecord return values but csvDatareader says "no header found"

### DISCUSSION THREAD

**Discussion by 1ly1mk** [JoshClose/CsvHelper#1255] [Q&A]
Hi,

I can the records mapped with my header in the file, but while creating datatable, csvDataReader says "No header found".
Any idea?

![image](https://user-images.githubusercontent.com/7780397/54042661-ad7b3480-41f0-11e9-889a-c058e0fb6415.png)

**Comment by augustoproiete** [comment]
@Mohakalilam Could you post a code sample that reproduces this issue?

**Comment by 1ly1mk** [comment]
```cs
using (var reader = new StreamReader(@"I:\FWorks\PropertyTax\AnokaCty.csv"))
                using (var csv = new CsvReader(reader))
                {
                    csv.Configuration.PrepareHeaderForMatch = (string header, int index) => Regex.Replace(header, @"\s", string.Empty);
                    csv.Configuration.HasHeaderRecord = true;
                    csv.Configuration.RegisterClassMap<OwnerMap>();
                    //csv.ReadAsync();
                    //csv.ReadHeader();

                    Owners = csv.GetRecords<Owner>().ToList();

                    using (var dr = new CsvDataReader(csv))
                    {
                        var dt = new DataTable();
                        dt.Load(dr);
                    }
                }
```

Also , I dont get the header using the ReadHeader , and so I had to comment those lines and I got the Records successfully. But now the issue is with datatable. 
Also on the csvhelper , I read that craeting a classmap is not required and we can handle using attribute but it doesnt work for me. Version I am using is 12.1.2.

**Comment by JoshClose** [comment]
Can you give the header row and the first data row?

What is the issue with `DataTable`?

How are you using the attributes?

**Comment by 1ly1mk** [comment]
Here is my header row

OwnerFirstName | OwnerLastName | PropAddress | PropCity | PropState | PropZip | MailingAddress | MailingCity | MailingState | MailingZip | Property County | CumulativeDue

The issue with datatable is that it returns null although I am able to get ercord using GetRecords().

**Comment by augustoproiete** [comment]
@Mohakalilam This happens because you are consuming the whole file with your call to `GetRecords<T>().ToList()` before loading the data table, which means your `StreamReader` is positioned at the very end of the file.

If you want to read the same stream again, you'll have to reset it, and create a new instance of your CsvReader (just to be safe, in case it does any kind of caching now or in future versions).

```
// ...
reader.BaseStream.Seek(0, SeekOrigin.Begin);  // <<<<<<< ******
reader.DiscardBufferedData();  // <<<<<<< ******

using (var dr = new CsvDataReader(new CsvReader(reader)))  // <<<<<<< ******
{
	var dt = new DataTable();
	dt.Load(dr);
}
```

Out of curiosity, why are you loading the CSV file twice? And why do you need to load into a `DataTable`?
